conda-store-server/conda_store_server/worker/tasks.py

```diff
@@ -62,21 +62,54 @@ def task_update_storage_metrics(self):
     )
 
 
+"""
+Pierre - May 29th 2022
+This is a different version of task_update_conda_channels.
+It's designed to run one task per channel, with a lock to avoid triggering twice a task for a given channel.
+The lock is handled by Redis.
+
+The reason behind having the task running only once at a time is
+to avoid integrity exceptions when running channel.update_packages.
+
+Sources :
+Lock : http://loose-bits.com/2010/10/distributed-task-locking-in-celery.html
+Redis : https://pypi.org/project/redis/
+https://stackoverflow.com/questions/12003221/celery-task-schedule-ensuring-a-task-is-only-executed-one-at-a-time
+
+"""
+
+
 @current_app.task(base=WorkerTask, name="task_update_conda_channels", bind=True)
 def task_update_conda_channels(self):
     conda_store = self.worker.conda_store
-    try:
-        conda_store.ensure_conda_channels()
 
-        for channel in api.list_conda_channels(conda_store.db):
+    conda_store.ensure_conda_channels()
+    for channel in api.list_conda_channels(conda_store.db):
+        send_task("task_update_conda_channel", args=[channel.name], kwargs={})
+
+
+@current_app.task(base=WorkerTask, name="task_update_conda_channel", bind=True)
+def task_update_conda_channel(self, channel_name):
+    conda_store = self.worker.conda_store
+
+    task_key = f"lock_{self.name}_{channel_name}"
+
+    is_locked = False
+    lock = conda_store.redis.lock(task_key)
```

Comment:
@pierrotsmnrd redis was removed as a requirement (due to trying to allow for a standalone mode). Is it possible for us to make this a non-requirement for the locking? I know that redis the the correct way to handle this...

Reply:
@costrouc  We absolutely need a lock of some sort, to avoid updating a channel twice at the same time to prevent from any error. The algorithm to populate the database is based on this assumption - which I guess is fine considering it's now very fast.

What other options do we have for a lock ?

Reply:
@pierrotsmnrd we could also use a filelock which for now will work fine. It will break down when we have multiple celery workers but in that case they should have redis anyways.